onnxruntime/core/providers/webnn/builders/impl/gru_op_builder.cc

```diff
@@ -0,0 +1,249 @@
+    split_options.set("label", node.Name() + "_split");
+    split_options.set("axis", 1);
+    // Split it to bias and recurrentBias.
+    emscripten::val splitted_biases =
+        model_builder.GetBuilder().call<emscripten::val>("split", bias, /*splits*/ 2, split_options);
+    options.set("bias", splitted_biases[0]);
+    options.set("recurrentBias", splitted_biases[1]);
+  }
+
+  if (input_defs.size() > 5 && input_defs[5]->Exists()) {
+    options.set("initialHiddenState", model_builder.GetOperand(input_defs[5]->Name()));
+  }
+
+  bool linear_before_reset = !!helper.Get("linear_before_reset ", 0);
+  options.set("resetAfter", linear_before_reset);
+
+  const auto& output_defs = node.OutputDefs();
+  bool has_Y = output_defs.size() > 0 && output_defs[0]->Exists();
+  bool has_Y_h = output_defs.size() > 1 && output_defs[1]->Exists();
+  options.set("returnSequence", has_Y);
+
+  std::string direction = helper.Get("direction", "forward");
+  if (direction == "forward") {
+    options.set("direction", emscripten::val("forward"));
+  } else if (direction == "reverse") {
+    options.set("direction", emscripten::val("backward"));
+  } else if (direction == "bidirectional") {
+    options.set("direction", emscripten::val("both"));
+  }
+
+  if (helper.HasAttr("activations")) {
+    const auto activations = helper.Get("activations", std::vector<std::string>{"Sigmoid", "Tanh"});
+    emscripten::val recurrent_network_activations = emscripten::val::array();
+    for (size_t i = 0; i < 2; ++i) {
+      const std::string& activation = activations[i];
+      if (activation == "Relu") {
+        recurrent_network_activations.call<void>("push", emscripten::val("relu"));
+      } else if (activation == "Sigmoid") {
+        recurrent_network_activations.call<void>("push", emscripten::val("sigmoid"));
+      } else if (activation == "Tanh") {
+        recurrent_network_activations.call<void>("push", emscripten::val("tanh"));
+      }
+    }
+
+    options.set("activations", recurrent_network_activations);
+  }
+
+  emscripten::val outputs = model_builder.GetBuilder().call<emscripten::val>("gru", input, weight, recurrent_weight,
+                                                                             steps, hidden_size, options);
+
+  if (has_Y) {
+    model_builder.AddOperand(output_defs[0]->Name(), outputs[1]);
+  }
+  if (has_Y_h) {
+    model_builder.AddOperand(output_defs[1]->Name(), outputs[0]);
+  }
+
+  return Status::OK();
+}
+
+bool GruOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& initializers, const Node& node,
+                                     const WebnnDeviceType /*device_type*/, const logging::Logger& logger) const {
+  const auto& input_defs = node.InputDefs();
+  if (input_defs.size() < 3) {
+    LOGS(logger, ERROR) << "GRU: input size must greater than or equal to 3";
+    return false;
+  }
+
+  std::vector<int64_t> input_shape;
+  if (!GetShape(*input_defs[0], input_shape, logger)) {
```

Comment:
```suggestion
  if (!GetShape(*input_defs[0], input_shape, logger) || input_shape.empty()) {
```
Same comment here - just as sanity check that the shape isn't empty (scalar) before accessing it on line 133 (due to malformed model).

Reply:
Done.